src/spi.rs

```diff
@@ -954,6 +954,64 @@ where
         self
     }
 
+    pub fn set_send_only(&mut self) {
+        self.spi
+            .cr1
+            .modify(|_, w| w.bidimode().set_bit().bidioe().set_bit());
+    }
+
+    fn is_send_only(&mut self) -> bool {
+        self.spi.cr1.read().bidimode().bit_is_set() && self.spi.cr1.read().bidioe().bit_is_set()
```

Comment:
We probably want to read it only once, it's a volatile read so the compiler won't be able to optimize it.